pkg/cortex/modules.go

```diff
@@ -439,9 +441,14 @@ func (t *Cortex) initTableManager() (services.Service, error) {
 
 	var extraTables []chunk.ExtraTables
 	if t.Cfg.PurgerConfig.Enable {
-		deleteStoreTableClient, err := storage.NewTableClient(t.Cfg.Storage.DeleteStoreConfig.Store, t.Cfg.Storage)
-		if err != nil {
-			return nil, err
+		var deleteStoreTableClient chunk.TableClient
```

Comment:
I don't think you should do this. I believe you should wrap the registerer here as well, to keep the delete store fully decoupled from the table manager.